docs/source/developer_guides/lora.md

```diff
@@ -152,6 +152,42 @@ The default LoRA settings in PEFT add trainable weights to the query and value l
 config = LoraConfig(target_modules="all-linear", ...)
 ```
 
+### LoRA+ optimized LoRA
+
+LoRA training can be optimized using [LoRA+](https://arxiv.org/abs/2402.12354), which uses different learning rates for the adapter matrices A and B, shown to increase finetuning speed by up to 2x and performance by 1-2%.
+
+Assuming you have a properly configured `PeftModel`, sub-class the `Trainer` replacing the `create_optimizer` method:
+
+```py
+# Snippet based on Axolotl's trainer_builder.py: https://github.com/axolotl-ai-cloud/axolotl/blob/985819d89bec921e919e7e83042a869f04a25974/src/axolotl/core/trainer_builder.py#L305-L384
+from transformers import Trainer
+from peft.optimizers import create_loraplus_optimizer
+
+class MyTrainer(Trainer):
+    def create_optimizer(self):
```

Comment:
Is this really necessary? I think it works equally well if we initialize the optimizer directly and pass it to the `Trainer` as argument. Also, that way, it's easier to adopt the example in case that users don't use `Trainer` at all.

```python

optimizer = create_loraplus_optimizer(...)
scheduler = None

...
trainer = Trainer(
    ...,
    optimizers=(optimizer, scheduler),
)
```

Reply:
That's way better, yeah. Thanks.